### ISSUE TITLE
`better-regex` enhancement: report invalid regexes (when exceptions thrown by the parser)

### ISSUE THREAD

**Issue by brettz9** [sindresorhus/eslint-plugin-unicorn#1081]
Invalid regexes (which can nevertheless pass through some parsers) are silently passed over despite the opportunity to report them.

`better-regex`

This should be reported even if the parser does not catch it. It is reported by the default `espree`, but not by `babel-eslint` or `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.

```js
const c = /(/;
```

(I've also filed https://github.com/DmitrySoshnikov/regexp-tree/issues/220 such that the following should also fail.)

```js
const c = /{/u;
```

There appear to be [other cases](https://github.com/DmitrySoshnikov/regexp-tree/search?q=throw) which throw as well.
